### PR TITLE
Unstoppable Domains

### PR DESCRIPTION
**Pull request by MeanBoyCousin**
# Task: Unstoppable Domains

## Description

- Display a user's unstoppable domain if they have one.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Document update
- [ ] Quality of life improvement
- [ ] Package upgrades

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have NATSPEC'd any new functions
- [x] If appropriate, I have properly tested my new functionality

### CODE REVIEW

File: packages/front-end/src/hooks/useUnstoppableDomain.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+export const useUnstoppableDomain = () => {
+  const { dispatch } = useGlobalContext();
+
+  const { address } = getAccount();
+
+  useEffect(() => {
+    const getDomain = async () => {
+      try {
+        if (address) {
+          const resolution = new Resolution();
```
Comment (devey) [inline packages/front-end/src/hooks/useUnstoppableDomain.ts:18]: can't this be done outside of `useUnstoppableDomain` ?

File: packages/front-end/src/hooks/useUnstoppableDomain.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+          const domain = await resolution.reverse(address);
+
+          if (domain) {
+            dispatch({
+              type: ActionType.SET_UNSTOPPABLE_DOMAIN,
+              unstoppableDomain:
+                domain.length > 24 ? `${domain.substring(0, 24)}...` : domain,
```
Comment (devey) [inline packages/front-end/src/hooks/useUnstoppableDomain.ts:26]: isn't it better if the three dots are in the middle ? doesn't matter much but was just thinking that Uniswap, at least for hex addresses, does it like that for example

**Review by devey — APPROVED** [review]
LGTM

File: packages/front-end/src/hooks/useUnstoppableDomain.ts (hunk 2 above)
Comment (MeanBoyCousin) [reply packages/front-end/src/hooks/useUnstoppableDomain.ts:26]: The main reason I did it like this is that this is how RainbowKit handles long ENS domains. Just wanted the continuity tbh. Hopefully one day they will get an integration for UD domains and then we can bin all this!

File: packages/front-end/src/hooks/useUnstoppableDomain.ts (hunk 1 above)
Comment (MeanBoyCousin) [reply packages/front-end/src/hooks/useUnstoppableDomain.ts:18]: Yeah probably makes more sense. I'll make it exportable as well for future-proofing.

File: packages/front-end/src/hooks/useUnstoppableDomain.ts (hunk 2 above)
Comment (devey) [reply packages/front-end/src/hooks/useUnstoppableDomain.ts:26]: sounds good, thanks

**Review by devey — APPROVED** [review]

**Pull request merged**